common/CommonAPI.h

```diff
@@ -734,23 +734,68 @@ class CommonAPI
 		uint32_t windowWidth = 800u;
 		uint32_t windowHeight = 600u;
 		uint32_t scImageCount = 3u;
-		
+
 		SFeatureRequest<nbl::video::IAPIConnection::E_FEATURE> requiredInstanceFeatures = {};
 		SFeatureRequest<nbl::video::IAPIConnection::E_FEATURE> optionalInstanceFeatures = {};
 		SFeatureRequest<nbl::video::ILogicalDevice::E_FEATURE> requiredDeviceFeatures = {};
 		SFeatureRequest<nbl::video::ILogicalDevice::E_FEATURE> optionalDeviceFeatures = {};
 		
-		nbl::asset::IImage::E_USAGE_FLAGS swapchainImageUsage = nbl::asset::IImage::E_USAGE_FLAGS::EUF_NONE;
-		nbl::core::vector<nbl::video::ISurface::SFormat> acceptableSurfaceFormats = {
-			nbl::video::ISurface::SFormat(nbl::asset::EF_R8G8B8A8_SRGB, nbl::asset::ECP_SRGB, nbl::asset::EOTF_sRGB),
-			nbl::video::ISurface::SFormat(nbl::asset::EF_B8G8R8A8_SRGB, nbl::asset::ECP_SRGB, nbl::asset::EOTF_sRGB)
+		nbl::asset::IImage::E_USAGE_FLAGS swapchainImageUsage = nbl::asset::IImage::E_USAGE_FLAGS::EUF_COLOR_ATTACHMENT_BIT;
```

Comment:
don use complex containters such as vectors, use pointer + count, two pointers and begin/end iterators or an invalid value terminated list

store the default values as `constexpr static inline std::array` or plain array